benchmark/Benchmark.h

```diff
@@ -0,0 +1,327 @@
+     *  Starts with (0,0).
+     * @param newEntryContent What to set the entry to.
+     */
+    template<typename T>
+      requires std::is_same<T, float>::value ||
+      std::is_same<T, std::string>::value
+    void setEntryOfExistingTable(const std::string& tableDescriptor,
+        const size_t row, const size_t column,
+        const T& newEntryContent) {
+      RecordTable::EntryType& entry =
+        getHashMapTableEntry(tableDescriptor, row, column);
+      entry = newEntryContent;
+    }
+
+    /*
+     * @brief Returns the content of a table entry, if the entry holds a value.
+     *
+     * @tparam T What type the entry has. Must be either float, or string. If
+     *  you give the wrong one, or the entry was never set/added, then this
+     *  function will cause an exception.
+     *
+     * @param tableDescriptor The identification of the table.
+     * @param row, column Where in the tables to write the measured time.
+     *  Starts with (0,0).
+     */
+    template<typename T>
+      requires std::is_same<T, float>::value ||
+      std::is_same<T, std::string>::value
+    const T getEntryOfExistingTable(const std::string& tableDescriptor,
+        const size_t row, const size_t column){
+      RecordTable::EntryType& entry =
+        getHashMapTableEntry(tableDescriptor, row, column);
+
+      // There is a chance, that the entry of the table does NOT have type T,
+      // in which case this will cause an error. As this is a mistake on the
+      // side of the user, we don't really care.
+      return std::get<T>(entry);
+    }
+
+    /*
+     * @brief Returns a vector of all the tables.
+     */
+    const std::vector<RecordTable> getTables() const;
+};
+
+/*
+ * This class exists as a round about way, of registering functions as
+ * benchmarks before entering the main function.
+ */
+class BenchmarkRegister {
+
+    // Alias for a type, so that we don't repeat things so often.
+    using BenchmarkFunction = std::function<void(BenchmarkRecords*)>;
+ 
+    /*
+     * @brief Returns a reference to the static vector of all registered
+     *  benchmark functions.
+     *  That is, the functions, that do the setup needed for a benchmark and
+     *  then record the execution time with the passed BenchmarkRecords.
+     *  The calling of the registered benchmark functions and the processing of
+     *  the recorded times, will be done in a main function later.
+     */
+    static std::vector<BenchmarkFunction>& getRegister();
+
+  public:
+
+    /*
+     * @brief Register one, or more, functions as benchmark functions
+     *  by creating a global instance of this class. Shouldn't take up much
+     *  space and I couldn't find a better way of doing it.
+     *
+     * @param benchmarks The functions can be passed as
+     *  `{&functionName1, &functionname2, ...}`. For more information about
+     *  their usage see getRegister . 
+     */
+    BenchmarkRegister(const std::vector<BenchmarkFunction>& benchmarks);
```

Comment:
Don't forget to adjust the main function, once you are done with this.

Reply:
Wrote interface in [8eaa1af](https://github.com/ad-freiburg/qlever/pull/860/commits/8eaa1af875f606841cb042a2330e469d3c6223f4).

Reply:
Integrated the basics of the interface in [867cdaa](https://github.com/ad-freiburg/qlever/pull/860/commits/867cdaaf965c79f02a642846330e4d9b5abc327c). The general metadata of the interface isn't used yet and their is no way to pass the configuration yet.

Reply:
Because not everybody is going to have an use for configurations and general metadata, I added some default behaviour in [42977f5](https://github.com/ad-freiburg/qlever/pull/860/commits/42977f56c0d9aca9c9bde0fc316bf9c5a4a5280a).

Reply:
Had a realisation, that the whole thing with passing the `BenchmarkRecords` to `runAllBenchmarks` was kinda stupid, if every class got there own version anyway. Why not have `runAllBenchmarks` create and return one itself, it would be far less management.
So I did that in [b26339e](https://github.com/ad-freiburg/qlever/pull/860/commits/b26339e013214514ef87972c5a7a3393a4006ef1).

Reply:
The general metadata of `BenchmarkClassInterface::getMetadata()` is now integrated, see [85aac9a](https://github.com/ad-freiburg/qlever/pull/860/commits/85aac9a8b49ca8f4dc26c758cccd6470c640ccbe). More specifically, the automatically written json file now includes it.